Provides a Taylor Diagram script

brianpm opened the pull request:
resolves #97

A new plotting script called `cam_taylor_diagram.py` is provided. It is designed to closely follow the AMWG version of the Taylor diagram. This includes providing calculation of the AMWG's "top 10" variables from the climo files. Those are hard-wired in, but would be easy to allow different variables in a future revision. 

No new dependencies, and no changes to the rest of ADF, so should not cause headaches. 

An example output plot is provided (just a screenshot, actually).

I didn't hook this up to get ingested into the website. I actually wasn't sure how to deal with that since these plot all cases on one plot. Maybe they should go in the same place as the table?
 
<img width="817" alt="taylor_prototype" src="https://user-images.githubusercontent.com/5471157/149242177-262b8212-c8ad-4579-9413-26be64b3dae5.png">

andrewgettelman commented:
Awesome Brian! Thanks for doing this. Looks good to me.

nusbaume reviewed (changes requested):
Looks good to me!  Just a few minor change requests.

Also, I can probably get this working in the website generator for both multi-case and single-case ADF runs, which would also help me lay the groundwork for future multi-variable and/or multi-case plots.  Thus if you want to hold off on my change requests then I can likely implement them myself along with the website changes sometime next week.

brianpm commented on scripts/plotting/cam_taylor_diagram.py:
```diff
@@ -0,0 +1,479 @@
+
+# --- DERIVED VARIABLES --- 
+
+def _pressure_from_hybrid(psfc, hya, hyb, p0=100000.):
```
Correct. Changed to use `pres_from_hybrid`.

brianpm commented on scripts/plotting/cam_taylor_diagram.py:
```diff
@@ -0,0 +1,479 @@
+        raise ValueError("No landfrac returned")
+    prect = get_prect(casename, location)
+    # mask to only keep ocean locations
+    prect = xr.DataArray(np.where(landfrac <= 0, prect, np.nan),
```
I think I saw someone else do this years ago and adopted it. I think using 0 or 0.05 wouldn't make much difference. I'll put it at 0.05 for now.

andrewgettelman replied:
Better probably to make it non-zero (that's what I always do). In case something gets very close to zero.

brianpm commented on scripts/plotting/cam_taylor_diagram.py:
```diff
@@ -0,0 +1,479 @@
+    return prect.sel(lat=slice(-30,30))
+
+
+def get_u_at_plev(adf, casename, location):
```
Generalized it a bit in new function `get_var_at_plev(adf, casename, location, variable, plev)` and made `get_u_at_plev(adf, casename, location)` a one-liner call to that one.

brianpm commented on scripts/plotting/cam_taylor_diagram.py:
```diff
@@ -0,0 +1,479 @@
+
+
+def _retrieve(adfobj, variable, casename, location, return_dataset=False):
+    """Custom function that retrieves a variable. Returns the variable as a DaraArray.
```
Done.

brianpm commented on scripts/plotting/cam_taylor_diagram.py:
```diff
@@ -0,0 +1,479 @@
+    mean_y = y.weighted(weights).mean()
+    dev_x = x - mean_x
+    dev_y = y - mean_y
+    dev_xy = dev_x * dev_y
+    cov_xy = dev_xy.weighted(weights).mean()
```
Yes, I agree. I'm not sure why I did it in two steps. Corrected.

brianpm commented:
@nusbaume -- I addressed all your comments and pushed them to my repo. I'm not sure if I'm supposed to "resolve conversation" or if you are supposed to.

nusbaume commented:
@brianpm thanks for the revisions!  I've gone ahead and made a separate issue for the website implementation, so I think this is good to go for merging.  

Also, in general I'll go ahead and resolve the conversations after checking that the new code modifications are what was expected (as that way I can make sure a new bug wasn't mistakenly added).

Thanks again for getting this necessary plot type developed!
